spec/forms/steps/conviction/conviction_length_form_spec.rb

```diff
@@ -0,0 +1,69 @@
+require 'spec_helper'
+
+RSpec.describe Steps::Conviction::ConvictionLengthForm do
+  let(:arguments) { {
+    disclosure_check: disclosure_check,
+    conviction_length: conviction_length,
+    conviction_length_type: conviction_length_type
+  } }
+  let(:disclosure_check) { instance_double(DisclosureCheck) }
+  let(:conviction_length) { 3 }
+  let(:conviction_length_type) { 'weeks' }
+
+  subject { described_class.new(arguments) }
+
+  describe '.choices' do
+    it 'returns the relevant choices' do
+      expect(described_class.choices).to eq(%w(
+        weeks
+        months
+      ))
+    end
+  end
+
+
+  describe '#save' do
+    context 'when form is valid' do
+      let(:conviction_length_type) { 'weeks' }
+
+      it_behaves_like 'a value object form', attribute_name: :conviction_length_type, example_value: 'weeks'
+
+
+      it 'saves the record' do
+        expect(disclosure_check).to receive(:update).with(
+          conviction_length_type: conviction_length_type,
+          conviction_length: conviction_length
+        ).and_return(true)
+
+        expect(subject.save).to be(true)
+      end
+    end
+
+    context 'Validation' do
+      context 'when conviction_length_type is not given' do
+        let(:conviction_length_type) { nil }
+
+        it 'returns false' do
+          expect(subject.save).to be(false)
+        end
+
+        it 'has a validation error on the field' do
+          expect(subject).to_not be_valid
```

Comment:
This is not testing the appropriate validation error has been added to the field.
Please refer to for an example: https://github.com/ministryofjustice/disclosure-checker/blob/master/spec/forms/steps/conviction/known_date_form_spec.rb#L34

Same for validations below.